xapian-letor/include/xapian-letor/ranker.h

```diff
@@ -193,7 +195,7 @@ class XAPIAN_VISIBILITY_DEFAULT Ranker : public Xapian::Internal::intrusive_base
   protected:
     /// Method to train the model. Overridden in ranker subclass.
     virtual void
-    train(const std::vector<Xapian::FeatureVector> & training_data) = 0;
+    train(const vector<vector<Xapian::FeatureVector>>& training_data) = 0;
```

Comment:
Probably not something to address right now, but I'm already luke warm on baking requirements to use a particular container into the API like we were here, and this change makes that situation worse.  Taking begin and end iterators for the data is a more flexible (and arguably a more natural C++ API since that's how it's done in the STL).